documentation.md

```diff
@@ -0,0 +1,521 @@
+
+We additionally introduce two address spaces:
+
+- _address_: refers to the address of the extended cells, viewing them as elementary storage elements. This matches with addresses as defined by the metadata RAM.
+- _full address_: refers to the full address of an elementary cell in the payload RAM.
+
+<figure class="image">
+  <img src="https://i.imgur.com/VzySGSn.png" alt="Simulated memory controller internal overview">
+  <figcaption>Fig: In-depth burst storage, for a burst length of 4</figcaption>
+</figure>
+
+<figure class="image">
+  <img src="https://i.imgur.com/fzaWZ6v.png" alt="Simulated memory controller internal overview">
+  <figcaption>Fig: In-depth burst storage addressing</figcaption>
+</figure>
+
+Therefore, there are _MaxBurstLen_ times as many elementary RAM cells in _i_payload_ram_ than there are in each of _i_meta_ram_out_tail_ and _i_meta_ram_out_head_.
+
+An extended cell is called _active_ if it has already acquired a response and has not acquired and released all the responses corresponding to the allocated burst.
+
+### Linked list implementation
+
+Linked lists enforce the ordering of the response release as defined by the AXI identifier.
+
+#### Linked list pointers
+
+Linked lists are logical structures maintained by the _i_meta_ram_out_tail_ and _i_meta_ram_out_head_ RAMs as well as four pointers:
+
+- Reservation head (_rsv_heads_q_): Points to the most recently reserved extended cell.
+- Response head (_rsp_heads_): Points to the next RAM address where a response of the corresponding AXI identifier will be stored.
+- Pre*tail (\_pre_tails*): Points to the second-to-last cell hosting or awaiting a response in the linked list.
+- Tail (_tails_): Points to the oldest cell hosting or awaiting a response in the linked list.
+
+The order of the pointers must always be respected. They can be equal but never overtake each other, in the order defined by the linked list.
+
+Two distinct tail pointers are required to dynamically manage the two following cases:
+
+- The pre_tail address is given as input to the payload RAM if there is a successful output handshake and the value at the output has an AXI id corresponding to the AXI id of the considered linked list. It must point to:
+  - The second-to-last element of the linked list if the list contains two reponses or more,
+  - The last element of the linked list if the list contains just one response,
+  - rsp_head if the list contains no responses.
+- The tail address is given as input to the payload RAM in all other cases. This case disjunction prevents an output data from being output twice, and prevents any bandwidth drop at the output. It must point to:
+
+  - The last element of the linked list if the list contains one response or more,
+  - pre_tail if the list contains no responses.
```

Comment:
I think it would be more clear if you would start this by saying that because of the read latency of the RAMs we need to apply the read address one cycle in advance to achieve maximum bandwidth. And then you can discuss how this is solved using the pre_tail and tail pointers.

Reply:
Okay! I added this!